### PR TITLE
[WIP] MoonViT implementation for multimodal Kimi K2.6 support

### PR DESCRIPTION
**Pull request by zbaczewskiTT**
### PR Category
Feature — MoonViT vision tower for Kimi-K2.6 multimodality support.

### Summary
Adds the MoonViT vision tower + multimodal projector for Kimi K2.6 as a `tt-metal/ttnn` implementation, validated against the HF torch reference. This is the MoonViT 400M vision encoder that turns image patches into 7168-dim tokens plus the host-side processor path.

Weights are loaded from moonshotai/Kimi-K2.6, vision-only: the tower and projector live entirely in shards 63–64 (~940 MB), so we load just those via safetensors.

Compute is bf16 with HiFi4 + fp32 dest-accumulation on all matmuls; Uses accurate GELU for PCC (~0.02) improvement at a negligible performance cost.

#### PCC vs fp32 reference

All inputs are uniform random RGB images run through the HF image processor
(decode → resize → normalize → patchify). PCC is vs the fp32 torch reference;
patch size is 14.

**Per-submodule** (threshold ≥ 0.99)

| Component | PCC |
|---|---|
| LayerNorm (norm0 / norm1 / final) | 0.99998 |
| MLP | 0.9998 |
| Attention | 0.99997 |
| Encoder block | 0.99985 |
| Projector | 0.99993 |

**Full pipeline** — tower + projector (threshold ≥ 0.95)

| Original resolution | Grid (patches) | Patches (L) | PCC |
|---|---|---|---|
| 224×224 | 16×16 | 256 | 0.986–0.991 |
| 336×224 | 24×16 | 384 | 0.990 |
| 336×336 | 24×24 | 576 | 0.977 |
| 448×448 | 32×32 | 1024 | 0.985 |
| 896×896 | 64×64 | 4096 | 0.981 |

#### Performance (Blackhole P100a, bf16)

| Original resolution | Patches (L) | Host processor (ms) | Device model (ms) | Total (ms) |
|---|---|---|---|---|
| 224×224 | 256 | 2.7 | 30.8 | 33.5 |
| 448×448 | 1024 | 14.2 | 52.2 | 66.4 |
| 672×672 | 2304 | 21.5 | 136.9 | 158.4 |
| 896×896 | 4096 | 41.0 | 340.5 | 381.5 |

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:zbaczewski/moonvit)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:zbaczewski/moonvit)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:zbaczewski/moonvit)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:zbaczewski/moonvit)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:zbaczewski/moonvit)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:zbaczewski/moonvit)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:zbaczewski/moonvit)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:zbaczewski/moonvit)

### CODE REVIEW

File: models/demos/deepseek_v3/tt/moonvit/_references.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    `max_seqlen` the 2D callers don't pass — we derive it from cu_seqlens.
+    Also forces the layer's attention to the requested implementation
+    (default `sdpa`, registered by `_k26_loader._register_sdpa_attention`,
+    which masks multi-image correctly unlike K2.6's bundled `eager`).
+    """
+
+    _INNER = "layer"
+
+    def __init__(self, layer):
+        super(_DelegatingModule, self).__init__()
```
[the remark this answers is not in the document]
Comment (zbaczewskiTT) [reply models/demos/deepseek_v3/tt/moonvit/_references.py]: this would crash;
changed to call grandparent explicitly though: `torch.nn.Module.__init__(self)`

File: models/demos/deepseek_v3/tt/moonvit/model.py [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+        HF Kimi-VL model. Loading happens lazily — first access to
+        `model_args.hf_model` triggers the checkpoint download (or hits
+        the local cache).
+        """
+        # Lazily resolve the HF model graph so the encoder.blocks list is in scope.
+        from models.demos.deepseek_v3.tt.moonvit._references import _vision_tower
+
+        vt = _vision_tower(model_args)
+        ref_blocks = vt.encoder.blocks
+        num_layers = len(ref_blocks)
```
[the remark this answers is not in the document]
Comment (zbaczewskiTT) [reply models/demos/deepseek_v3/tt/moonvit/model.py]: fix applied
